ntlm.go

```diff
@@ -198,6 +217,84 @@ func ntlmSessionResponse(clientNonce [8]byte, serverChallenge [8]byte, password
 	return response(hash, passwordHash)
 }
 
+func ntlmHashNoPadding(val string) []byte {
+	hash := make([]byte, 16)
+	h := md4.New()
+	h.Write(utf16le(val))
+	h.Sum(hash[:0])
+
+	return hash
+}
+
+func hmacMD5(passwordHash, data []byte) []byte {
+	hmacEntity := hmac.New(md5.New, passwordHash)
+	hmacEntity.Write(data)
+
+	return hmacEntity.Sum(nil)
+}
+
+func getTargetInformationData(bytes []byte) (infoBytes []byte, target string, err error) {
+	type2MessageLength := len(bytes)
+	targetNameAllocated := binary.LittleEndian.Uint16(bytes[14:16])
+	targetNameOffset := binary.LittleEndian.Uint32(bytes[16:20])
+	if type2MessageLength < int(targetNameOffset + uint32(targetNameAllocated)) {
+		return nil, target, errorInvalidServerType2Message
+	}
+
+	target, err = decodeUTF16le(bytes[targetNameOffset:targetNameOffset+uint32(targetNameAllocated)])
+	if err != nil {
+		return nil, target, err
+	}
+
+	targetInformationAllocated := binary.LittleEndian.Uint16(bytes[42:44])
+	targetInformationDataOffset := binary.LittleEndian.Uint32(bytes[44:48])
+	if type2MessageLength < int(targetInformationDataOffset + uint32(targetInformationAllocated)) {
+		return nil, target, errorInvalidServerType2Message
+	}
+
+	targetInformationBytes := make([]byte, targetInformationAllocated)
+	copy(targetInformationBytes, bytes[targetInformationDataOffset:targetInformationDataOffset+uint32(targetInformationAllocated)])
+
+	return targetInformationBytes, target, nil
+}
+
+func getNTLMV2Response(target, username, password string, challenge, nonce [8]byte, targetInformation []byte, timestamp time.Time) []byte {
```

Comment:
A bit of docs on this function would be great. Especially references to a spec. (no need to add docs within the function, but I have no way to know what it is expected to do right now)

Reply:
I added references to official microsoft documentation, and an unofficial walk through which imho explains everything well. Also i added a reference to the specific place in the unofficial walk through that touches upon the contents of LMv2 and NTLMv2, in the spot where i create those payloads.